### DISCUSSION TITLE
Is the OTBR-agent expected to exit when the FactoryReset and Reset commands are sent via dBus?

### DISCUSSION THREAD

**Discussion by romacdon** [openthread/openthread#9808] [Q&A] [answered]
We have noticed that the OTBR-agent exits when the FactoryReset and Reset commands are sent via dBus but not when using ot-ctl. Is this expected?

To reproduce build the OTBR-agent on a pi then send the following command: 

sudo dbus-send --reply-timeout=30000 --print-reply --system --type="method_call" --dest=io.openthread.BorderRouter.wpan0 /io/openthread/BorderRouter/wpan0 io.openthread.BorderRouter.FactoryReset

 This causes the otbr to exit. Meanwhile, using ot-ctl to send the command succeeds.

**Comment by wgtdkp** [comment]
I don't this is expected, do you have the otbr-agent log for the d-bus "factoryreset" test?

**Reply by romacdon** [reply]
See attached log. Command sent at 18:08:47 & 18:09:09

[OTBR_exit_dbus.log](https://github.com/openthread/openthread/files/14072843/OTBR_exit_dbus.log)

**Reply by wgtdkp** [reply, marked as answer]
```
Jan 26 18:08:47 raspberrypi otbr-agent[1802]: 01:00:34.607 [C] Platform------: CPC init failed. Ensure radio-url argument has the form 'spinel+cpc://cpcd_0?iid=<1..3>'
Jan 26 18:08:47 raspberrypi otbr-agent[1802]: 01:00:34.607 [C] Platform------: Init() at cpc_interface.cpp:96: Failure
```

d-bus reset and ot-ctl reset are different implementations, and I think it's silicon labs code that doesn't handle the d-bus reset implementation well (e.g. it doesn't support re-init). There is a similar issue in home assistant though not related to d-bus reset command: https://github.com/home-assistant/addons/issues/3192

**Comment by romacdon** [comment]
> I don't this is expected, do you have the otbr-agent log for the d-bus "factoryreset" test?



> ```
> Jan 26 18:08:47 raspberrypi otbr-agent[1802]: 01:00:34.607 [C] Platform------: CPC init failed. Ensure radio-url argument has the form 'spinel+cpc://cpcd_0?iid=<1..3>'
> Jan 26 18:08:47 raspberrypi otbr-agent[1802]: 01:00:34.607 [C] Platform------: Init() at cpc_interface.cpp:96: Failure
> ```
> 
> d-bus reset and ot-ctl reset are different implementations, and I think it's silicon labs code that doesn't handle the d-bus reset implementation well (e.g. it doesn't support re-init). There is a similar issue in home assistant though not related to d-bus reset command: [home-assistant/addons#3192](https://github.com/home-assistant/addons/issues/3192)

Thanks.  I missed that in the log.  I will investigate further and thanks for the link to the other issue.

**Comment by romacdon** [comment]
This issue appears to be in the restart logic for the Silabs CPC code.  We will continue to investigate internally.
